tests/connect.phpt

```diff
@@ -7,7 +7,7 @@ pdo_snowflake.cacert=libsnowflakeclient/cacert.pem
     include __DIR__ . "/common.php";
 
     // full parameters
-    $dbh = new PDO("$dsn;application=phptest;authenticator=snowflake;priv_key_file=tests/p8test.pem;priv_key_file_pwd=test", $user, $password);
+    $dbh = new PDO("$dsn;application=phptest;authenticator=snowflake;priv_key_file=tests/p8test.pem;priv_key_file_pwd=test;disablequerycontext=true;includeretryreason=false", $user, $password);
```

Comment:
should we have the tests also with the other combination of the new connection parameters?

Reply:
@sfc-gh-dprzybysz  htap test case added. No visible result can be verified in the test case itself so it can only be verified through driver log.
[snowflake_20230922112020.txt](https://github.com/snowflakedb/pdo_snowflake/files/12703725/snowflake_20230922112020.txt)
From the log you can see in the first connection with default settings, the parameters passed to libsnowflakeclient is correct:
```
2023-09-22 18:20:20.459 DEBUG C     client.c          518: qcc_disable: false
2023-09-22 18:20:20.459 DEBUG C     client.c          519: include_retry_reason: true
```
And you can see the query context cache works as expected (up to 4 entries back and forth)

While in the second connection with non-default settings, he parameters passed to libsnowflakeclient is correct as well
```
2023-09-22 18:20:28.305 DEBUG C     client.c          518: qcc_disable: true
2023-09-22 18:20:28.305 DEBUG C     client.c          519: include_retry_reason: false
```
and no QueryContextDTO in the following query requests.

No confirmation on the behavior of retry reason but since it's been tested on libsnowflakeclient, as long as the parameter passed to libsnowflakeclient correctly it should be fine.